cmd/cstorvolumeclaim/cstorvolumeclaim.go

```diff
@@ -342,26 +319,26 @@ func getOrCreateCStorVolumeResource(
 // on the available cstor pools created for storagepoolclaim.
 // if pools are less then desired replicaCount its return an error.
 func distributeCVRs(
-	replicaCount int,
+	desiredReplicaCount int,
+	claim *apis.CStorVolumeClaim,
 	service *corev1.Service,
 	volume *apis.CStorVolume,
-	class *storagev1.StorageClass,
 ) error {
 
-	spcName := getSPC(class)
+	spcName := getSPC(claim)
 	if len(spcName) == 0 {
-		return errors.New("failed to get spc name from storageClass")
+		return errors.New("failed to get spc name from cstorvolumeclaim")
 	}
 
-	poolList, err := listCStorPools(spcName, replicaCount)
+	poolList, err := listCStorPools(spcName, claim.Spec.ReplicaCount)
 	if err != nil {
 		return err
 	}
 
 	uniquePoolList := getUniquePoolList(volume.Name, poolList)
 	for i, pool := range uniquePoolList.Items {
 		pool := pool
-		if i < replicaCount {
+		if i < desiredReplicaCount {
 			_, err = creatCVR(service, volume, &pool)
```

Comment:
function name is wrong.. should be createCVR?

Reply:
`desiredReplicaCount` gives the number of CVRs that need to be created sparsely..
so, createCVR should return obj and nil as error only if it is created.. so, createCVR should return error if error in creation or already exists..

Reply:
shouldn't return from the error happened with createCVR